src/main/java/nl/topicus/mssql2monetdb/util/MonetDBUtil.java

```diff
@@ -435,44 +440,91 @@ public static void dropMonetDBTableOrView (MonetDBTable monetDBTable) throws SQL
 	public static void dropAndRecreateViewForTable(String schema, String name,
 			MonetDBTable monetDBTable) throws SQLException
 	{
-		String fullName = schema + "." + name;
+		Connection conn = CopyToolConnectionManager.getInstance().getMonetDbConnection();
+		
+		// disable autocommit to work in a single transaction
+		boolean oldAutoCommit = conn.getAutoCommit();
+		conn.setAutoCommit(false);
+		
+		// make sure the referenced table exists 
 		if (monetDBTableExists(monetDBTable))
 		{
+			String fullName = schema + "." + name;
+			
 			LOG.info("Drop and recreate or create the view '" + fullName + "' for table '"
 				+ monetDBTable.getToTableSql() + "' on MonetDB server...");
-
-			try
-			{
-				Statement stmt =
-					CopyToolConnectionManager.getInstance().getMonetDbConnection()
-						.createStatement();
-				StringBuilder builder = new StringBuilder();
-				if (tableOrViewExists(schema, name))
+			
+			// drop current table/view
+			boolean alreadyDropped = false;
+			while(tableOrViewExists(schema, name))
```

Comment:
misschien nog een max aantal trys zodat je niet in een eeuwig loop komt mocht het toch zo zijn dat de tabel of view niet te droppen is door wat voor reden dan ook